### PR TITLE
Set the minimum Java version to 16

### PR DESCRIPTION
**Pull request by Camotoy**
This removes compatibility with all Java versions prior to 16, as Minecraft 1.17 now requires Java 16 and the Minecraft community is slowly shifting to a 16 base.

This PR should not be merged until there is complete agreement among the core developers that it is the right time for the move.

(Also Bedrock 1.17 support is dropped)

### CODE REVIEW

**Comment by lukeeey** [comment]
You’ve gotta change the pull request workflow to use java 16 which is why the build is failing

**Review by RednedEpic — APPROVED** [review]

**Review by DoctaEnkoda — APPROVED** [review]

**Review by lukeeey — APPROVED** [review]

**Review by Tim203 — APPROVED** [review]

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 1]
```diff
@@ -137,7 +137,6 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
                                 break;
                             }
                         }
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:140]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 2]
```diff
@@ -155,19 +154,15 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
...
+                                FALL_FLYING, // Elytra
+                                SPIN_ATTACK -> // Trident spin attack
                                 playerPosition = playerPosition.sub(0, (EntityType.PLAYER.getOffset() - 0.4f), 0);
-                                break;
-                            case SLEEPING:
+                            case SLEEPING ->
                                 playerPosition = playerPosition.sub(0, (EntityType.PLAYER.getOffset() - 0.2f), 0);
-                                break;
                         } // else, we don't have to modify the position
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:170]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 3]
```diff
@@ -198,15 +193,13 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
                                 break;
                             }
                         }
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:201]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 4]
```diff
@@ -198,15 +193,13 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
...
                         ClientPlayerPlaceBlockPacket blockPacket = new ClientPlayerPlaceBlockPacket(
                                 new Position(packet.getBlockPosition().getX(), packet.getBlockPosition().getY(), packet.getBlockPosition().getZ()),
                                 BlockFace.values()[packet.getBlockFace()],
                                 Hand.MAIN_HAND,
                                 packet.getClickPosition().getX(), packet.getClickPosition().getY(), packet.getClickPosition().getZ(),
                                 false);
                         session.sendDownstreamPacket(blockPacket);
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:209]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 5]
```diff
@@ -234,7 +227,6 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
                                 }
                             }
                         }
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:237]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 6]
```diff
@@ -248,15 +240,14 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
...
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:251]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 7]
```diff
@@ -283,14 +274,12 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
                                 break;
                             }
                         }
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:286]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 8]
```diff
@@ -283,14 +274,12 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
...
                         ClientPlayerUseItemPacket useItemPacket = new ClientPlayerUseItemPacket(Hand.MAIN_HAND);
                         session.sendDownstreamPacket(useItemPacket);
-                        break;
-                    case 2:
+                    }
+                    case 2 -> {
                         int blockState = session.getGameMode() == GameMode.CREATIVE ?
                                 session.getConnector().getWorldManager().getBlockAt(session, packet.getBlockPosition()) : session.getBreakingBlock();
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:293]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 9]
```diff
@@ -301,37 +290,34 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
...
-                        diffZ = playerPosition.getZ() - (floatBlockPosition.getZ() + 0.5f);
+                        float diffX = playerPosition.getX() - (floatBlockPosition.getX() + 0.5f);
+                        float diffY = (playerPosition.getY() - EntityType.PLAYER.getOffset()) - (floatBlockPosition.getY() + 0.5f) + 1.5f;
+                        float diffZ = playerPosition.getZ() - (floatBlockPosition.getZ() + 0.5f);
                         float distanceSquared = diffX * diffX + diffY * diffY + diffZ * diffZ;
                         if (distanceSquared > MAXIMUM_BLOCK_DESTROYING_DISTANCE) {
                             restoreCorrectBlock(session, packet.getBlockPosition(), packet);
                             return;
                         }
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:314]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 10]
```diff
@@ -301,37 +290,34 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
...
                         LevelEventPacket blockBreakPacket = new LevelEventPacket();
                         blockBreakPacket.setType(LevelEventType.PARTICLE_DESTROY_BLOCK);
                         blockBreakPacket.setPosition(packet.getBlockPosition().toFloat());
                         blockBreakPacket.setData(session.getBlockMappings().getBedrockBlockId(blockState));
                         session.sendUpstreamPacket(blockBreakPacket);
                         session.setBreakingBlock(BlockStateValues.JAVA_AIR_ID);
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:321]: Revert

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 11]
```diff
@@ -301,37 +290,34 @@ public void translate(GeyserSession session, InventoryTransactionPacket packet)
...
                         Entity itemFrameEntity = ItemFrameEntity.getItemFrameEntity(session, packet.getBlockPosition());
                         if (itemFrameEntity != null) {
                             ClientPlayerInteractEntityPacket attackPacket = new ClientPlayerInteractEntityPacket((int) itemFrameEntity.getEntityId(),
                                     InteractAction.ATTACK, session.isSneaking());
                             session.sendDownstreamPacket(attackPacket);
                             break;
                         }
-
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:329]: Revert

File: connector/src/main/java/org/geysermc/connector/skin/FloodgateSkinUploader.java [hunk 12]
```diff
@@ -132,18 +132,10 @@ public void onMessage(String message) {
...
-                                    break;
-                                case 1:
-                                    logger.error("Got a message from skin uploader: " + logMessage);
-                                    break;
-                                default:
-                                    logger.info(logMessage);
-                                    break;
+                                case -1 -> logger.debug("Got a message from skin uploader: " + logMessage);
+                                case 0 -> logger.info("Got a message from skin uploader: " + logMessage);
+                                case 1 -> logger.error("Got a message from skin uploader: " + logMessage);
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/skin/FloodgateSkinUploader.java:137]: These can be collated
```suggestion
                                case -1, 0, 1 -> logger.debug("Got a message from skin uploader: " + logMessage);
```

**Review by rtm516 — DISMISSED** [review]

**Pull request merged**